Makefile

```diff
@@ -115,6 +115,8 @@ manifests: controller-gen
 	yq d -i config/crd/serving.kubeflow.org_inferenceservices.yaml 'spec.versions[1].schema.openAPIV3Schema.properties.spec.properties.*.properties.*.properties.livenessProbe.properties.httpGet.required'
 	yq d -i config/crd/serving.kubeflow.org_inferenceservices.yaml 'spec.versions[1].schema.openAPIV3Schema.properties.spec.properties.*.properties.*.properties.readinessProbe.properties.tcpSocket.required'
 	yq d -i config/crd/serving.kubeflow.org_inferenceservices.yaml 'spec.versions[1].schema.openAPIV3Schema.properties.spec.properties.*.properties.*.properties.livenessProbe.properties.tcpSocket.required'
+	cp -u config/crd/serving.kubeflow.org_inferenceservices.yaml test/crds/serving.kubeflow.org_inferenceservices.yaml
```

Comment:
Why do we need to copy the crd files?

Reply:
If there's other files(here it's kustomize) an error will be reported. But I think we need not copy CRD files because we can specify to path level through Environment.CRDInstallOptions

Reply:
Yeah the issue is about [envtest](https://book.kubebuilder.io/reference/envtest.html) as it now errors if you point it at a kustomize dir. In master it points [at the config dir](https://github.com/kubeflow/kfserving/pull/1144/files#diff-68abf77fa6ead6f002a59acab8552c391eaf159cdfe897c83096ddae128f30e6L54) and in the new version envtest tries to load [kustomization.yaml](https://github.com/kubeflow/kfserving/blob/master/config/crd/kustomization.yaml) and errors that it doesn't contain a Kind. I can change the envtest path but it needs a path, it can't be pointed at single files. And the path with the crds does contain that kustomization. I figured that since it's already pointing at the [test directory](https://github.com/kubeflow/kfserving/tree/4bddfbc0b650261cb867d50512149a07e37f7fab/test/crds) (which also contains istio crds) it would make sense to just copy the built crds to there every time they're built and keep that test directory up to date and make it the [only place that the tests load CRDs from](https://github.com/kubeflow/kfserving/pull/1144/files#diff-5cefe84774ca5f18960f72fe12ad5e8f7722b1ba535ab68402c797837ce7385dR40)

Reply:
Thanks, I understand since I encounter the same error and must move kustomize.yaml out to make it work. But after investigation I found when setting up envtest there's CRDInstallOptions can be passed and can point to file:
https://github.com/kubernetes-sigs/controller-runtime/blob/bae8fdb711031aa10dbf7361dbe77a5a650aa8d9/pkg/envtest/crd.go#L44

It's said this can take effect together with CRDDirectoryPaths:
https://github.com/kubernetes-sigs/controller-runtime/blob/bae8fdb711031aa10dbf7361dbe77a5a650aa8d9/pkg/envtest/server.go#L111-L114

Reply:
Did you manage to get it to work with files? I tried it and found I got an error as in the controller-runtime [code block that tries to handle it](https://github.com/kubernetes-sigs/controller-runtime/blob/bae8fdb711031aa10dbf7361dbe77a5a650aa8d9/pkg/envtest/crd.go#L304) the if and the else both assume a Dir. I can try again.

Reply:
Yes, though there's error due to test case failed that I did not get time to figure out, but at least the install CRD part works for me with below code in envtest_setup.go:
```
func SetupEnvTest() *envtest.Environment {
	t := &envtest.Environment{
		CRDInstallOptions: envtest.CRDInstallOptions{
			Paths: []string{
				filepath.Join("..", "..", "..", "..", "..", "..", "config", "crd", "serving.kubeflow.org_inferenceservices.yaml"),
				filepath.Join("..", "..", "..", "..", "..", "..", "config", "crd", "serving.kubeflow.org_trainedmodels.yaml"),
				filepath.Join("..", "..", "..", "..", "config", "crd", "serving.kubeflow.org_inferenceservices.yaml"),
				filepath.Join("..", "..", "..", "..", "config", "crd", "serving.kubeflow.org_trainedmodels.yaml"),
			},
		},
		// The relative paths must be provided for each level of test nesting
		// This code should be illegal
		CRDDirectoryPaths: []string{
			//filepath.Join("..", "..", "..", "..", "..", "..", "config", "crd"),
			filepath.Join("..", "..", "..", "..", "..", "..", "test", "crds"),
			//filepath.Join("..", "..", "..", "..", "config", "crd"),
			filepath.Join("..", "..", "..", "..", "test", "crds"),
		},
		UseExistingCluster: proto.Bool(false),
	}
```

Reply:
@pugangxa you were right, I made a mistake before. It does work with files. Have changed so no need for a copy now. Thanks for the tip.